skrifa/src/color/instance.rs

```diff
@@ -0,0 +1,660 @@
+            for (i, (ty, delta)) in tys.iter().zip(deltas.iter_mut()).enumerate() {
+                let var_index = var_index_base + i as u32;
+                // If we don't have a var index map, use our index as the inner
+                // component and set the outer to 0.
+                let delta_ix = DeltaSetIndex {
+                    outer: 0,
+                    inner: var_index as u16,
+                };
+                *delta = compute_delta(ty, delta_ix);
+            }
+        }
+        deltas
+    }
+}
+
+#[derive(Copy, Clone)]
+enum DeltaType {
+    FWord,
+    UfWord,
+    Fixed,
+    F2Dot14,
+}
+
+impl<'a> Deref for ColrInstance<'a> {
+    type Target = Colr<'a>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.colr
+    }
+}
+
+/// Resolves a clip box, applying variation deltas using the given
+/// instance.
+pub fn resolve_clip_box(instance: &ColrInstance, clip_box: &ClipBox) -> BoundingBox<f32> {
+    match clip_box {
+        ClipBox::Format1(cbox) => BoundingBox {
+            x_min: cbox.x_min().to_i16() as f32,
+            y_min: cbox.y_min().to_i16() as f32,
+            x_max: cbox.x_max().to_i16() as f32,
+            y_max: cbox.y_max().to_i16() as f32,
+        },
+        ClipBox::Format2(cbox) => {
+            let deltas = instance.var_deltas(cbox.var_index_base(), [DeltaType::FWord; 4]);
+            BoundingBox {
+                x_min: cbox.x_min().apply_delta(deltas[0]),
+                y_min: cbox.y_min().apply_delta(deltas[1]),
+                x_max: cbox.x_max().apply_delta(deltas[2]),
+                y_max: cbox.y_max().apply_delta(deltas[3]),
+            }
+        }
+    }
+}
+
+/// Simplified version of a [`ColorStop`] or [`VarColorStop`] with applied
+/// variation deltas.
+#[derive(Clone, Debug)]
+pub struct ResolvedColorStop {
+    pub offset: f32,
+    pub palette_index: u16,
+    pub alpha: f32,
+}
+
+/// Collection of [`ColorStop`] or [`VarColorStop`].
+// Note: only one of these fields is used at any given time, but this structure
+// was chosen over the obvious enum approach for simplicity in generating a
+// single concrete type for the `impl Iterator` return type of the `resolve`
+// method.
+pub struct ColorStops<'a> {
+    stops: &'a [ColorStop],
+    var_stops: &'a [VarColorStop],
+}
+
+impl<'a> ColorStops<'a> {
+    pub fn len(&self) -> usize {
+        self.stops.len() + self.var_stops.len()
+    }
+}
+
+impl<'a> From<ColorLine<'a>> for ColorStops<'a> {
+    fn from(value: ColorLine<'a>) -> Self {
+        Self {
+            stops: value.color_stops(),
+            var_stops: &[],
+        }
+    }
+}
+
+impl<'a> From<VarColorLine<'a>> for ColorStops<'a> {
+    fn from(value: VarColorLine<'a>) -> Self {
+        Self {
+            stops: &[],
+            var_stops: value.color_stops(),
+        }
+    }
+}
+
+impl<'a> ColorStops<'a> {
+    /// Returns an iterator yielding resolved color stops with variation deltas
+    /// applied.
+    pub fn resolve(
```

Comment:
I think given that I see some differences in the offsets in my results before and after this change, it would be useful to add a few tests for this part, just to be able to verify this one level lower down.

Reply:
Will do!

Reply:
Fixed the intermediate 2.14 overflow and updated the IVS floating point test to check 2.14 and 16.16 deltas.